osgar/tools/lidarview.py

```diff
@@ -313,6 +311,16 @@ def _step(self, direction):
                 self.scan2 = deserialize(data)
             elif stream_id == self.camera_id:
                 self.image = get_image(deserialize(data))
+                current = self.current
+                while current + direction >= 0 and current + direction < len(self.log):
+                    current += direction
```

Comment:
does it really work as expected for negative direction? I expected (from PR info), that bbox is always **after** the image? BTW is it within the "time frame", i.e. no images are dropped?

Reply:
I know, right? I am surprised that it works too :) Seriously though, I was hoping someone who is familiar with the overall architecture would chime in, I didn't dig deep enough but is it possible that the viewer does some pre-computation from beginning to the end first and then it just uses the precomputed values as you scroll left and right along the timeline?

Reply:
How do you know "it works"? For me it looks like when you step back you get the bboxes before the image and not after it. We have the same problem elsewhere (like when stepping back and looking at scan generated from depth image etc.) and somehow live with it. Maybe we can just acknowledge the problem exists and move on since we have lived with it forever now.

The viewer (lidarview.py) does no precomputation - it just scrolls back and forth through the logfile.

Reply:
To me it looked correct going both forward and backward (compared to before when it looked incorrect and one frame off going both forward and backward). We can certainly apply another fix if it turns out it's off after all.

Compared to other streams where the exact sync is not that important as the data simply passed through the system around the same time the image was taken but this data is not directly related to the image, here the bounding box needs to be displayed on top of the correct image or its informational value goes down dramatically (you typically want to find out what was the object/bounding box that was identified in this image).